### PR TITLE
[Composer] Fix value object inliner with Version

### PR DESCRIPTION
**Pull request by TomasVotruba**
Mirror of https://github.com/rectorphp/rector/pull/5175, with fix

### CODE REVIEW

File: rules/composer/src/ValueObject/ComposerModifier/AddPackageToRequire.php [hunk 1]
```diff
@@ -20,23 +19,18 @@ final class AddPackageToRequire implements ComposerModifierInterface
...
 
     /**
-     * @var Version
+     * @var string
      */
     private $version;
 
-    /**
-     * @param string $packageName name of package (vendor/package)
-     * @param string $version target package version (1.2.3, ^1.2, ~1.2.3 etc.)
```
Comment (lulco) [inline rules/composer/src/ValueObject/ComposerModifier/AddPackageToRequire.php:29]: Do you really want to remove this?

File: rules/composer/src/ValueObject/ComposerModifier/AddPackageToRequireDev.php [hunk 2]
```diff
@@ -20,23 +19,18 @@ final class AddPackageToRequireDev implements ComposerModifierInterface
...
     private $version;
 
-    /**
-     * @param string $packageName name of package (vendor/package)
-     * @param string $version target package version (1.2.3, ^1.2, ~1.2.3 etc.)
-     */
     public function __construct(string $packageName, string $version)
     {
         $this->packageName = $packageName;
-        $this->version = new Version($version);
```
Comment (lulco) [inline rules/composer/src/ValueObject/ComposerModifier/AddPackageToRequireDev.php:34]: now there is no validation, what about at least call `new Version($version);` at first line to validate version string?

**Review by lulco — COMMENTED** [review]
The final version will look like one of my version 7 days ago :P

File: rules/composer/tests/Processor/Fixture/composer_before.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+    "require": {
+        "nette/nette": "^3.0"
```
Comment (lulco) [inline rules/composer/tests/Processor/Fixture/composer_before.json:3]: this should be 2.4 here

File: rules/composer/src/ValueObject/ComposerModifier/AddPackageToRequire.php (hunk 1 above)
Comment (TomasVotruba) [reply rules/composer/src/ValueObject/ComposerModifier/AddPackageToRequire.php:29]: Yea, these are obvious comments that do not any value.

File: rules/composer/src/ValueObject/ComposerModifier/AddPackageToRequireDev.php (hunk 2 above)
Comment (TomasVotruba) [reply rules/composer/src/ValueObject/ComposerModifier/AddPackageToRequireDev.php:34]: WIP :) validation should be in 1 place.

File: rules/composer/tests/Processor/Fixture/composer_before.json (hunk 3 above)
Comment (TomasVotruba) [reply rules/composer/tests/Processor/Fixture/composer_before.json:3]: Good catch! Already changed in next PR.

**Comment by TomasVotruba** [comment]
> The final version will look like one of my version 7 days ago :P

It's possible :D give me an hour

**Pull request merged**